pelicun/uq.py

```diff
@@ -1638,18 +1718,16 @@ def inverse_transform(self, values: np.ndarray) -> np.ndarray:
           Inverse CDF values
 
         """
-        theta, beta = self.theta[:2]
+        theta, truncation_limits = self._prepare_arrays(values)
+        theta, beta = theta.T
 
         if np.any(~np.isnan(self.truncation_limits)):
-            a, b = self.truncation_limits
+            a, b = truncation_limits.T
 
-            if np.isnan(a):
-                a = np.nextafter(0, 1)
-            else:
-                a = np.maximum(np.nextafter(0, 1), a)
-
-            if np.isnan(b):
-                b = np.inf
+            # Replace NaN values
+            a = np.nan_to_num(a, nan=np.nextafter(0, 1))
+            a[a == 0] = np.nextafter(0, 1)
```

Comment:
I think we should check for `a<=0` instead. Any negative `a` will be just as bad as a zero `a`.